src/driver/src/DefaultDriver.cpp

```diff
@@ -87,6 +88,67 @@ void reportMetrics(genny::metrics::Registry& metrics,
     actorSetup.report(std::move(finishTime), std::move(duration), std::move(outcome));
 }
 
+/**
+ * Filter to remove any nodes inside dedicated "ignore" values.
+ *
+ * Basically re-implementing the following python logic:
+ *  ignored = set("foo","bar")
+ *  return [u for u in unused if not any(ignore in u for ignore in ignored)]
+ */
+UnusedNodes removeIgnored(const UnusedNodes& unused, const std::vector<std::string>& ignored) {
+    UnusedNodes out{};
+    std::copy_if(
+        unused.begin(), unused.end(), std::back_inserter(out), [&](const std::string& path) {
+            return !std::any_of(ignored.begin(), ignored.end(), [&](const std::string& ignore) {
+                return path.find(ignore) != std::string::npos;
+            });
+        }
+    );
+    return out;
+}
+
+/**
+ * Log a message showing the unused yaml structures.
+ */
+void reportUnused(const NodeSource& nodeSource, const bool dryrun) {
+    auto raw = nodeSource.unused();
+
+    std::vector<std::string> ignored{".yml/Description", ".yml/Owner", ".yml/AutoRun", ".yml/Keywords"};
+    if (dryrun) {
+        ignored.push_back(".yml/Clients");
+    }
+
+    auto unused = removeIgnored(raw, ignored);
+    auto many = std::distance(unused.begin(), unused.end());
+    auto verb = many == 1 ? "was" : "were";
+    auto plural = many == 1 ? "" : "s";
+    auto action = dryrun ? "constructing" : "running";
+    std::stringstream message;
+    message << std::endl << "<BETA FEATURE> - YAML Usage Check" << std::endl;
+    
+    if (!unused.empty()) {
+        message << "There " << verb << " " << many << " YAML structure" << plural
+                            << " unused when " << action << " the workload." << std::endl;
+        message << "\n\t" << boost::algorithm::join(unused, "\n\t") << std::endl;
+        
+    } else {
+        message << "All YAML structures appear to have been used when " << action
+                << " this workload." << std::endl;
+    }
+    message << "The following nodes were ignored in this analysis:" << std::endl;
+    message << "\n\t" << boost::algorithm::join(ignored, "\n\t") << std::endl;
+
+    message << "Incorrect results are possible. "
```

Comment:
Shouldn't this be displayed only if unused is not empty?

Reply:
The incorrect results are possible part? I think that could be the case either way? e.g. maybe unused is empty and we were wrong about that?